server/src/server.ts

```diff
@@ -425,6 +407,49 @@ connection.onCompletion((pos: CompletionParams): CompletionItem[] => {
     return completionProvider.getAvailableCompletionItems(pos.textDocument.uri, doc.offsetAt(pos.position));
 });
 
+connection.onDocumentFormatting((_params) => {
+    const doc = documents.get(_params.textDocument.uri);
+    // We should directly send an empty list if this feature is disabled by the settings.
```

Comment:
I struggled a lot with code coverage. I'm not sure if adding this condition now will make code coverage OK.
Just in case, I prefer to let the code like this and add a condition later.